### PR TITLE
v2: remove deprecated targetport

### PR DESCRIPTION
**Pull request by hannahkm**
### What does this PR do?

Remove the `TargetPort` constant, which was deprecated in PR #2016. Replace instances with `NetworkDestinationPort`.

### Motivation

For v2 release, we should remove support for deprecated constants and functions, especially those that have been deprecated for many, subsequent minor versions.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: ddtrace/ext/tags.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ const (
 
 	// TargetPort sets the target host port.
 	// Deprecated: Use NetworkDestinationPort instead.
-	TargetPort = "out.port"
```
Comment (rarguelloF) [inline ddtrace/ext/tags.go:23]: Hmm I don't think this are the kind of breaking changes we should do in v2, as these could break stuff in the users monitors, dashboards, etc. - unless we decide is ok because no one is using these.

(also, in case we decide to go forward with this change, this constant should be entirely removed instead of the string value being changed, as this could be really confusing)

**Review by rarguelloF — CHANGES_REQUESTED** [review]

**Comment by hannahkm** [comment]
TL;DR from offline conversation: we will keep support for `TargetPort` (and similar constants) and remove the deprecated label. Customers should still use `NetworkDestinationPort` over `TargetPort` when possible.
